rfc-028-adopt-a-code-of-conduct.md

```diff
@@ -0,0 +1,138 @@
+# Adopt a code of conduct for public repositories
+
+## Summary
+
+All dxw-owned public repositories MUST have a code of conduct and a guide to
+contributing.
+
+All repositories created by dxw which will be transferred to a client SHOULD
+have a code of conduct and a guide to contributing, if feasible.
+
+## Problem
+
+When we publish software which allows people outside dxw to submit questions,
+issues or contributions, we’re implicitly creating a community.
+
+Anyone who chooses to join this community should feel safe to interact with it
+and able to reveal any aspect of their identity that they want to without
+suffering abuse.
+
+They need to know that there are standards of behaviour which apply to them and
+the others in the community. They need to be able to report violations of these
+standards to the community leaders.
+
+## Proposal
+
+We should adopt a code of conduct, so that people know how they can expect to
+be treated when they interact with our projects – hopefully making them feel
+safe to do so.
+
+The most common code of conduct is the [Contributor
+Covenant](https://www.contributor-covenant.org/). It pledges to make
+participation in an open source project a harrassment-free experience for
+everyone. It explains:
+
+- the standards of behaviour which are expected of everyone who interacts with
+  the project
+- who’s responsible for enforcing the standards
+- how they’ll enforce the standards – the consequences of violating the
+  standards
+
+We want to limit the scope of who finds out about a code of conduct report, to
+maintain privacy. So we’ll limit it to the Chief Technology Officer and
+Managing Director.
+
+### Guidelines
+
+All public repositories in the `dxw` organisation MUST contain a `CONTRIBUTING.md` and `CODE_OF_CONDUCT.md` file.
+
+The `CONTRIBUTING.md` file:
+
+- SHOULD explain whether the project is actively seeking contributions from
+  outside dxw
+- MUST link to the `CODE_OF_CONDUCT.md` file
+
+The `CODE_OF_CONDUCT.md` file:
+
+- SHOULD be a copy of the [Markdown version of the Contributor Covenant](https://www.contributor-covenant.org/version/2/0/code_of_conduct/).
```

Comment:
There's a CLI to automate some of this, if you think that's relevant: https://github.com/simonv3/covenant-generator

Reply:
That's an option, although if we do the "Create fallback CONTRIBUTING.md and CODE_OF_CONDUCT.md files for the dxw GitHub organisation" step then we’ll just be able to copy it from dxw's `.github` repository, might be easier.